src/sentry/api/endpoints/organization_repository_details.py

```diff
@@ -49,13 +50,21 @@ def put(self, request, organization, repo_id):
 
         serializer = RepositorySerializer(data=request.DATA, partial=True)
 
-        if serializer.is_valid():
-            result = serializer.object
-            if result.get('status'):
-                if result['status'] in ('visible', 'active'):
-                    repo.update(status=ObjectStatus.VISIBLE)
-                else:
-                    raise NotImplementedError
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        result = serializer.object
+        update_kwargs = {}
+        if result.get('status'):
+            if result['status'] in ('visible', 'active'):
+                update_kwargs['status'] = ObjectStatus.VISIBLE
+            else:
+                raise NotImplementedError
```

Comment:
I think we should be raising an more API-relevant error here – probably a 422 (or 400 to match the validation case above). This will just end up with a generic "500: Internal Error".

Reply:
this was there before: https://github.com/getsentry/sentry/pull/9978/files#diff-6e3a10ff3614a2910f4149aa6e5abe1aL58 so while i don't disagree, i think it's out of scope for this pr.